mobilelib.php

```diff
@@ -235,7 +256,7 @@ function get_allowed_tag_users($forum_id=0, $group_id=0, $advancedforum=0, $repl
             JOIN {role_assignments} ra ON (u.id = ra.userid AND ra.contextid = ?)
             JOIN {role} r ON (ra.roleid = r.id)
             WHERE e.courseid = ?
-            AND r.shortname IN ('coursecoach', 'headtutor', 'tutor', 'support')
+            AND r.shortname != 'student'
```

Comment:
Do we need to worry about any potential roles derived from the student archetype?

Reply:
Not in this case. All the other queries in this file target 'student' directly where this query needs to excluded it and work for all other roles. Most mentions work targets the archetypes, then the other roles are specifically targeted where needed.